### ISSUE TITLE
error: namespaces "jx-staging" not found

### ISSUE THREAD

**Issue by cheesydev** [jenkins-x/jx#6153]
### Summary 
In a fresh Jenkins X installation, after changing the environment with the command `jx env` and selecting 'staging', the `jx` command becomes pretty much unusable, printing `error: namespaces "jx-staging" not found` when attempting to run other commands.

### Steps to reproduce the behavior

1. Create a new cluster in GKE with all default settings
2. Run `jx boot` in an empty directory and finish installation
3. Run `jx env` and select 'staging'
4. Run `jx env` again (or pretty much any other `jx` sub command). The command prints `error: namespaces "jx-staging" not found` and fails.

### Expected behavior
In step 4 above, jx commands should work properly, e.g. `jx env` should list existing environments.

### Actual behavior
Command fails printing `error: namespaces "jx-staging" not found`.

### Jx version

Before getting stuck in the problematic internal state, the output of `jx version` was:

```
NAME               VERSION
jx                 2.0.1001
Kubernetes cluster v1.13.11-gke.14
kubectl            v1.14.1
helm client        Client: v2.14.1+g5270352
git                2.20.1
Operating System   Debian GNU/Linux 10 (buster)
```

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster
GKE, 3 n1-standard-1 nodes, all default settings.

### Operating system / Environment
Debian 10 (buster)

`uname -a`: 4.19.0-6-amd64 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux

**Comment by deanesmith** [comment]
Hi @cheesydev if you add a quickstart or a new project it should resolve.  Please confirm since they are created lazily.

**Comment by deanesmith** [comment]
Considering having boot establish the environments initially.  Advantage is for secrets replication.

**Comment by ram4444** [comment]
similar error when using jx create quickstart when it comes to promotion to staging
Could not find the service URL in namespace jx-staging for names testspring2, jx-testspring2, jx-staging-testspring2

I am a bit confused since the only namespace created is jx but no jx-staging jx-production

is modifying env/jxboot-resources/values.tmpl.yaml will make it work?
or I should do something else before jx boot?

**Comment by valiksb** [comment]
Having same issue 
`NAME               VERSION
jx                 2.0.1258
Kubernetes cluster v1.15.9-gke.26
kubectl            v1.17.4
helm client        2.16.5
git                2.24.1 (Apple Git-126)
Operating System   Mac OS X 10.15.4 build 19E266`

Any updates?

**Comment by johnamac7** [comment]
so I've moved my JX install across now away from the Jenkins pipelines to use the JX Tekton. I've been able to create a project but when the pipeline runs I get the same error as above.

From the project pipeline logs:

Pull Request https://github.com/user1/environment-testcluster-staging/pull/1 is merged at sha a960fb718a70629de93f9807a4888ddd00f4da1f
Pull Request merged but we are not waiting for the update pipeline to complete!
WARNING: Could not find the service URL in namespace jx-staging for names initial-test-proj, jx-initial-test-proj, jx-staging-initial-test-proj

I have 3 environments defined:
'''
12:15 $ jx get env
NAME       LABEL       KIND        PROMOTE NAMESPACE     ORDER CLUSTER SOURCE                                                              REF    PR
dev        Development Development Never   jx            0             https://github.com/johnamac7/environment-testcluster-dev.git        master
staging    Staging     Permanent   Auto    jx-staging    100           https://github.com/johnamac7/environment-testcluster-staging.git    master
production Production  Permanent   Manual  jx-production 200           https://github.com/johnamac7/environment-testcluster-production.git master
'''


Is there an update as to why this is the case. In the old system the staging pipeline would have been kicked off after the merge was complete and the project would have completed.

It would be great to have this resolved.
